File: algorithms/solutions/java/_0283_MoveZeroes.java
package hot100;

public class _0283_MoveZeroes {
// @lc code=start
    class Solution {
        public void moveZeroes(int[] nums) {
            int left = 0, right = 0;
            while (right < nums.length){
                if (nums[right] != 0){
                    if (left != right){
                        swap(nums, left, right);
                    }
                    left++;
                }
                right++;
            }
        }

        private void swap(int[] nums, int i, int j){
            int t = nums[i];
            nums[i] = nums[j];
            nums[j] = t;
        }
    }
// @lc code=end
}

File: algorithms/solutions/python3/0283_move_zeroes.py
#
# @lc app=leetcode.cn id=283 lang=python3
#
# [283] Move Zeroes
#
# https://leetcode.cn/problems/move-zeroes/description/
#

# @lc code=start
from typing import List


class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        left = right = 0
        while right < len(nums):
            if nums[right] != 0:
                if left != right:
                    nums[left], nums[right] = nums[right], nums[left]
                left += 1
            right += 1
                    
        
# @lc code=end

